include/devicetree/pm_state.h

```diff
@@ -0,0 +1,165 @@
+/*
+ * Copyright (c) 2020 Intel corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#ifndef ZEPHYR_INCLUDE_DEVICETREE_PM_STATE_H_
+#define ZEPHYR_INCLUDE_DEVICETREE_PM_STATE_H_
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include <power/power_state.h>
+
+/**
+ * @defgroup devicetree-pm_states Power Management states device tree macros
+ * @{
+ */
+
+/**
+ * @brief Construct a pm_state_info from 'pm_states' property at index 'i'
+ *
+ * @param node_id A node identifier compatible with pm-states
+ * @param i index of pm_states prop which type is 'phandles'
+ * @return pm_state_info item from 'pm_states' property at index 'i'
+ */
+#define DT_PM_STATE_INFO_ITEM_BY_IDX(node_id, i)                    \
+	{                                                           \
+		.state = DT_ENUM_IDX(DT_PHANDLE_BY_IDX(node_id,     \
+					pm_states, i), pm_state),    \
+		.min_residency_us = DT_PROP_BY_PHANDLE_IDX_OR(node_id, \
+				pm_states, i, min_residency_us, -1),   \
+	},
+
+/**
+ * @brief Length of 'pm-states' property which type is 'phandles'
+ *
+ * @param node_id A node identifier compatible with pm-states
+ * @return length of 'pm-states' property which type is 'phandles'
+ */
+#define DT_PM_STATE_ITEMS_LEN(node_id) DT_PROP_LEN(node_id, pm_states)
+
+/**
+ * @brief Macro function to construct enum pm_state item in UTIL_LISTIFY
+ * extension.
+ *
+ * @param child child index in UTIL_LISTIFY extension.
+ * @param node_id A node identifier compatible with pm-states
+ * @return macro function to construct a pm_state_info
+ */
+#define DT_PM_STATE_INFO_ITEMS_LISTIFY_FUNC(child, node_id) \
+	DT_PM_STATE_INFO_ITEM_BY_IDX(node_id, child)
+
+/**
+ * @brief Macro function to construct a list of 'pm_state_info' items by
```

Comment:
Same, this and other structure initializer macro magic should move to the header that defines the structure.

Reply:
Initially I'd added these macros in the file that define the struct. Now I see two possibilities,

1) Remove devicetree/pm_states.h and move these macros to power/power_state.h and change macros namespace.
2) Remove the `#include <power/power_state.h>` from `devicetree/pm_state.h` since any macro is explicitly referencing things from that file.

The option 1 is my favorite.

Reply:
> Remove devicetree/pm_states.h and move these macros to power/power_state.h and change macros namespace.

no objections!

Reply:
great, will proceed this way.